### PR TITLE
Placing units_building on a grid an spawning them according to placements (networked)

### PR DESCRIPTION
**Pull request by GreeFine**

### CODE REVIEW

**Comment by GreeFine** [comment]
Added Engame text/actions. Nexus receiving damage and Ressouce creation/display/usage and better camera management

File: Assets/Scripts/Builder.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        ghost = null;
+        if (isServer)
+          CmdPlaceNewBuilding(PlayerController.pId, "default", pos);
+        else if (buyUnit(PlayerController.pId, "default"))
+          CmdPlaceNewBuilding(PlayerController.pId, "default", pos);
+      }
+    }
+  }
+
+  private bool buyUnit(int pId, string buildingName)
```
Comment (MaiDaine) [inline Assets/Scripts/Builder.cs]: typo

File: Assets/Scripts/CameraController.cs [hunk 2]
```diff
@@ -4,21 +4,47 @@
...
+  private Vector3 startPos;
 
-    void Update()
+  private Vector3 startPosP2;
+
+  public void Init()
+  {
+    if (PlayerController.pId == 1)
+    {
+      this.transform.position += new Vector3(128, 0, 0);
```
Comment (MaiDaine) [inline Assets/Scripts/CameraController.cs]: And'hur est satisfait ;)

File: Assets/Scripts/Nexus.cs [hunk 3]
```diff
@@ -5,10 +5,22 @@
...
 public class Nexus : Tower
 {
-    public int pPidOwner;
-    public Image healthBar;
+  public int pPidOwner;
+
+  protected override void Start()
+  {
+    MaxHealth = 30;
+    health = 10; //FIXME
```
Comment (MaiDaine) [inline Assets/Scripts/Nexus.cs]: Pourquoi pas les avoir un public au dessus pour regler depuis l editor

Comment (GreeFine) [reply Assets/Scripts/Nexus.cs]: It was to make test quicker but i removed it

**Pull request merged**